test/BeforeLinkAppendHook.test.js

```diff
@@ -0,0 +1,49 @@
+import MiniCssExtractPlugin from '../src';
+
+import { getCompiler, compile } from './helpers';
+
+it('It should be able to inject successfully', async (done) => {
+  class RewriteErrorHandler {
+    // eslint-disable-next-line class-methods-use-this
+    apply(compiler) {
+      compiler.hooks.thisCompilation.tap(
+        'RewriteErrorHandler',
+        (compilation) => {
+          compilation.hooks.miniCssExtractPluginBeforeLinkAppend.tap(
+            'RewriteErrorHandler',
+            (source, chunk, hash) => {
+              expect(source).toBeDefined();
+              expect(chunk).toBeDefined();
+              expect(hash).toBeDefined();
+              return 'head.onerror = console.error;';
```

Comment:
Why do you need rewrite error handle? What is real use case this hook?

Reply:
The actual use case is that I need to listen for network error events and initiate a retry，Domain names may be switched during the process.

Reply:
> The actual use case is that I need to listen for network error events and initiate a retry

Reply:
I can get a normal error, but I want to set a timeout for the resource load, then switch the domain name and reload it.
If I don't pass the hook, It will be difficult to get the start loading time of the resource.

Reply:
What is the weird use case? Why do you need this? It is potential break cache